Inline code snippets from the samples

cdupuis opened the pull request:
This PR adds a transfrom that inlines the follwing code snippet reference from the samples repo:

```
<!-- atomist:code-snippet:start=lib/sdm/dotnetCore.ts#dotnetGenerator -->
Anything can be in here
<!-- atomist:code-snippet:end-->
```

Once this is placed inside a markdown file in the docs repo, it will replace the section with the contents from the `lib/sdm/dotnetCore.ts`. See https://github.com/atomist/samples/blob/master/lib/sdm/dotnetCore.ts#L36-L53 for how the sections are marked in the samples.

cdupuis committed:
Inline code snippets from the samples

This PR adds a transfrom that inlines the follwing code snippet
reference from the samples repo:

```
<!-- atomist:code-snippet:start=lib/sdm/dotnetCore.ts#dotnetGenerator -->
Anything can be in here
<!-- atomist:code-snippet:end-->
```

Once this is placed inside a markdown file in the docs repo, it will
replace the section with the contents from the `lib/sdm/dotnetCore.ts`.
See https://github.com/atomist/samples/blob/master/lib/sdm/dotnetCore.ts#L36-L53
for how the sections are marked in the samples.

jessitron commented:
Cool, thank you! 
Tomorrow when I'm streaming, I'll test a few error cases and then merge it

Commit:
Return failure if a snippet is missing

Commit:
Now it's time to include a warning in a comment

Commit:
Note editedness

Commit:
Autofix: TypeScript imports

[atomist:generated] [atomist:autofix=typescript_imports]

Commit:
Implement explicit messages for Sample File Not Found.

This also removes the dependency on a different repo in the tests.
and adds a test with two files in it.

Commit:
Remove commented code

Commit:
When replacement is successful, add a link to the source

cdupuis commented:
Pull request rebase is in progress because @jessitron pushed 1 commit to **master**:
- 69b13a7 _Test codeowners_

Commit:
Formatting of source links

Commit:
Lint

cdupuis commented:
Pull request auto merged by Atomist.

* 1 approved review by @jessitron
* 2 successful checks

[atomist:generated] [auto-merge:on-approve]
